src/Table/Grid.php

```diff
@@ -411,7 +411,7 @@ protected function buildColumn($name, $callback = null)
      */
     protected function buildModel($model)
     {
-        if ($this->paginate === true && method_exists($model, 'paginate')) {
+        if ($this->isEloquentModel($model) && $this->paginate) {
```

Comment:
I would say it might be better to convert $model to query builder before this line.

```php
if ($this-> isEloquentModel($model)) {
    $model = $model->newQuery();
}

if ($this->paginate === true && method_exists($model, 'paginate')) {
   //
```